loopy/statistics.py

```diff
@@ -582,54 +637,73 @@ def subst_into_to_count_map(space, tcm, subst_dict):
 
 # {{{ CountGranularity
 
-class CountGranularity:
-    """Strings specifying whether an operation should be counted once per
+class CountGranularity(Enum):
+    """Specify whether an operation should be counted once per
     *work-item*, *sub-group*, or *work-group*.
 
     .. attribute:: WORKITEM
 
-       A :class:`str` that specifies that an operation should be counted
-       once per *work-item*.
+       Specifies that an operation should be counted once per *work-item*.
 
     .. attribute:: SUBGROUP
 
-       A :class:`str` that specifies that an operation should be counted
-       once per *sub-group*.
+       Specifies that an operation should be counted once per *sub-group*.
 
     .. attribute:: WORKGROUP
 
-       A :class:`str` that specifies that an operation should be counted
-       once per *work-group*.
+       Specifies that an operation should be counted once per *work-group*.
 
     """
 
-    WORKITEM = "workitem"
-    SUBGROUP = "subgroup"
-    WORKGROUP = "workgroup"
-    ALL: ClassVar[Sequence[str]] = [WORKITEM, SUBGROUP, WORKGROUP]
+    WORKITEM = 0
+    SUBGROUP = 1
+    WORKGROUP = 2
 
 # }}}
 
 
 # {{{ Op descriptor
 
-class Op(ImmutableRecord):
+class OpType(Enum):
```

Comment:
Move to `autoattribute` and in-line docstrings?

Reply:
I'm not sure - does adding in-line docstrings add a lot of information to these values? Also, with `autoattribute`, sphinx shows the integer values of these constants, which I think isn't very meaningful.

![image](https://github.com/user-attachments/assets/aeb3caa6-32be-475b-a8f1-0ee9dc1fbebd)